MantidQt/MantidWidgets/src/DataProcessorUI/GenericDataProcessorPresenter.cpp

```diff
@@ -520,53 +521,104 @@ std::string GenericDataProcessorPresenter::getPostprocessedWorkspaceName(
   return prefix + boost::join(outputNames, "_");
 }
 
-/**
-Loads a run from disk or fetches it from the AnalysisDataService
-@param run : The name of the run
-@param instrument : The instrument the run belongs to
-@param prefix : The prefix to be prepended to the run number
-@throws std::runtime_error if the run could not be loaded
-@returns a shared pointer to the workspace
-*/
+/** Loads a run found from disk or AnalysisDataService
+ *
+ * @param run : The name of the run
+ * @param instrument : The instrument the run belongs to
+ * @param prefix : The prefix to be prepended to the run number
+ * @throws std::runtime_error if the run could not be loaded
+ * @returns a shared pointer to the workspace
+ */
 Workspace_sptr
-GenericDataProcessorPresenter::loadRun(const std::string &run,
-                                       const std::string &instrument,
-                                       const std::string &prefix) {
+GenericDataProcessorPresenter::getRun(const std::string &run,
+                                      const std::string &instrument,
+                                      const std::string &prefix) {
+
+  bool runFound;
+  std::string outName;
+  std::string fileName = instrument + run;
+
+  outName = findRunInADS(run, prefix, runFound);
+  if (!runFound ||
+      AnalysisDataService::Instance().retrieveWS<IEventWorkspace>(outName) !=
+          NULL) {
+    // Workspace must not be an event workspace
```

Comment:
I think one of the major problems I ran into while doing this was that when loading runs from the ADS, it was perfectly possible for `GenericDataProcessorPresenter` to load a run with the right name, but was an event workspace which is unusable by this class. A way this could happen was if we used set some slicing values (thus using `ReflDataProcessorPresenter` and loading runs as event workspaces), then remove slicing values (using `GenericDataProcessorPresenter`) which would attempt to use the event workspace loaded previously in the ADS. I believe this was a problem before I worked on this issue too.